### PR TITLE
[WIP] New default Android presenter

### PR DESCRIPTION
**Pull request by martijn00**
## :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
It introduces a new Android presenter that is able to handle Fragments in any sort of Activity, support for Dialogs, and optional attributes for Activities.

## :arrow_heading_down: What is the current behavior?
No support for any of that.

## :new: What is the new behavior (if this is a feature change)?
Support for the new features:

- [x] Attributes for: Activtiy, Fragment, Tab, Dialog
- [x] Showing of those views even without Attribute
- [x] CustomAnimations per Attribute
- [x] SharedElement transitions
- [x] No dependency on MvxCachingFragmentActivity. All Activities can show Fragments now
- [x] Support for Fragments in default non AppCompat presenter
- [x] Closing of Dialog, Fragments, and Fragments when Activity closes
- [x] Override behaviour on runtime with `IMvxOverridePresentationAttribute`
- [x] Navigation to Tabs in ViewPager
- [x] Nested Fragments
- [ ] Caching of fragments from presenter
- [ ] Handle tombstoning
- [ ] Refactor `MvxCachingFragmentPagerAdapter` and `MvxCachingFragmentStatePagerAdapter`
- [ ] Load ViewModels for Fragments, Tabs, Dialogs, etc from within them
- [ ] Cleanup ViewModel cache connected to Fragments (`MvxFragmentExtensions`, `MvxFragmentExtensionMethods`)
- [ ] Support dynamic ViewPager adapter
- [ ] Cleanup `MvxViewPagerFragment`

Reproduction samples needed:
- [x] Navigate from `AppStart`
- [x] Navigate to `Activity`
- [x] Navigate to `Fragment` in the current open `Activity`
- [x] Navigate to `Fragment` in another `Activity`
- [x] Navigate to ` Fragment` with multiple Attributes
- [x] Navigate to `ViewPager Fragment`
- [x] Navigate to `TabLayout Fragment`
- [ ] Navigate to an already open `ViewPagerFragment`
- [x] Use a nested `Fragment` with `ChildFragmentManager`
- [x] Navigate to a `DialogFragment`
- [ ] Navigate to new `Fragment` within the `DialogFragment`
- [x] Use `IMvxOverridePresentationAttribute`
- [ ] Use custom non-standard `Attribute`
- [ ] Use SharedElements to do a Activity and Fragment transition
- [x] Close support for all Attributes
- [x] Master-detail example

## :boom: Does this PR introduce a breaking change?
Yes, but for apps most things should be easy to change.

## :bug: Recommendations for testing
Fragments and navigation between them.

## :memo: Links to relevant issues/docs
https://github.com/MvvmCross/MvvmCross/issues/1934

## :thinking: Checklist before submitting
This PR continues from https://github.com/MvvmCross/MvvmCross/pull/2022

- [x] All projects build
- [x] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [ ] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [x] Nuspec files were updated (when applicable)
- [x] Rebased onto current develop

### CODE REVIEW

**Comment by martijn00** [comment]
@Nickolas- @nmilcoff @orzech85 @andyci I've opened this PR from a branch here. This makes it possible to either push or PR directly to it.

File: MvvmCross-AndroidSupport/MvvmCross.Droid.Support.Fragment/MvxViewPagerFragmentInfo.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        public MvxViewPagerFragmentInfo(string title, Type fragmentType, Type viewModelType, object parameterValuesObject = null)
+            : this(title, null, fragmentType, viewModelType, parameterValuesObject)
+        {
+        }
+
+        public MvxViewPagerFragmentInfo(string title, string tag, Type fragmentType, Type viewModelType,
+                            object parameterValuesObject = null)
+        {
+            Title = title;
+            Tag = tag ?? title;
```
Comment (Cheesebaron) [inline MvvmCross-AndroidSupport/MvvmCross.Droid.Support.Fragment/MvxViewPagerFragmentInfo.cs]: I think this needs to be processed a bit. Meaning I don't think we are allowed to put anything inside of a tag. But I might be wrong.

File: MvvmCross/Droid/Droid/Views/Attributes/MvxAndroidPresentationAttributeExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+using MvvmCross.Platform.Droid.Platform;
+
+namespace MvvmCross.Droid.Views.Attributes
+{
+    public static class MvxAndroidPresentationAttributeExtensions
+    {
+        private static Type GetActivityViewModelType(Type activityType)
+        {
+            IMvxViewModelTypeFinder associatedTypeFinder;
+            if (!Mvx.TryResolve(out associatedTypeFinder))
```
Comment (Cheesebaron) [inline MvvmCross/Droid/Droid/Views/Attributes/MvxAndroidPresentationAttributeExtensions.cs]: I'd like to invert this to make it a bit easier on the eyes.

File: MvvmCross/Droid/Droid/Views/Attributes/MvxDialogFragmentPresentationAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+namespace MvvmCross.Droid.Views.Attributes
+{
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = true)]
+    public class MvxDialogFragmentPresentationAttribute : MvxFragmentPresentationAttribute
+    {
+        public MvxDialogFragmentPresentationAttribute()
+        {
+        }
+
+        public MvxDialogFragmentPresentationAttribute(bool cancelable = true, Type activityHostViewModelType = null, bool addToBackStack = false, int enterAnimation = int.MinValue, int exitAnimation = int.MinValue, int popEnterAnimation = int.MinValue, int popExitAnimation = int.MinValue, int transitionStyle = int.MinValue, bool isCacheableFragment = false) : base(activityHostViewModelType, int.MinValue, addToBackStack, enterAnimation, exitAnimation, popEnterAnimation, popExitAnimation, transitionStyle, null, isCacheableFragment)
```
Comment (Cheesebaron) [inline MvvmCross/Droid/Droid/Views/Attributes/MvxDialogFragmentPresentationAttribute.cs]: Can we put some breaks in here, the ctor lines are suuuuuper long. My screen is not wide enough.

File: MvvmCross/Droid/Droid/Views/Attributes/MvxTabLayoutPresentationAttribute.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+namespace MvvmCross.Droid.Views.Attributes
+{
+    [AttributeUsage(AttributeTargets.Class)]
+    public class MvxTabLayoutPresentationAttribute : MvxViewPagerFragmentPresentationAttribute
+    {
+        public MvxTabLayoutPresentationAttribute()
+        {
+        }
+
+        public MvxTabLayoutPresentationAttribute(string title, int viewPagerResourceId, int tabLayoutResourceId, Type activityHostViewModelType = null, bool addToBackStack = false, Type fragmentHostViewType = null, bool isCacheableFragment = false) : base(title, viewPagerResourceId, activityHostViewModelType, addToBackStack, fragmentHostViewType, isCacheableFragment)
```
Comment (Cheesebaron) [inline MvvmCross/Droid/Droid/Views/Attributes/MvxTabLayoutPresentationAttribute.cs]: Long lines again

File: MvvmCross/Droid/Droid/Views/Attributes/MvxViewPagerFragmentPresentationAttribute.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+namespace MvvmCross.Droid.Views.Attributes
+{
+    [AttributeUsage(AttributeTargets.Class)]
+    public class MvxViewPagerFragmentPresentationAttribute : MvxFragmentPresentationAttribute
+    {
+        public MvxViewPagerFragmentPresentationAttribute()
+        {
+        }
+
+        public MvxViewPagerFragmentPresentationAttribute(string title, int viewPagerResourceId, Type activityHostViewModelType = null, bool addToBackStack = false, Type fragmentHostViewType = null, bool isCacheableFragment = false) : base(activityHostViewModelType, int.MinValue, addToBackStack, int.MinValue, int.MinValue, int.MinValue, int.MinValue, int.MinValue, fragmentHostViewType, isCacheableFragment)
```
Comment (Cheesebaron) [inline MvvmCross/Droid/Droid/Views/Attributes/MvxViewPagerFragmentPresentationAttribute.cs]: Long lines

**Pull request merged**